draft-ietf-oauth-selective-disclosure-jwt.md

```diff
@@ -274,70 +291,176 @@ The array is created as follows:
+  in the `address` claim can only be disclosed in full. Other options are
+  discussed in (#nested_data).
+
+The Issuer creates the following Disclosures:
+
+{{examples/simple/disclosures.md}}
+
+The SD-JWT is then signed by the Issuer to create a JWT like the following:
+
+<{{examples/simple/sd_jwt_serialized.txt}}
+
+## Decoy Digests {#decoy_digests}
+
+An Issuer MAY add additional digests to the SD-JWT that are not associated with
+any claim.  The purpose of such "decoy" digests is to make it more difficult for
+an attacker to see the original number of claims contained in the SD-JWT. Decoy
+digests MAY be added both to the `_sd` array for objects as well as in arrays.
+
+It is RECOMMENDED to create the decoy digests by hashing over a
+cryptographically secure random number. The bytes of the digest MUST then be
+base64url-encoded as above. The same digest function as for the Disclosures MUST
+be used.
 
-#### Nested Data in SD-JWTs {#nested_data}
+For decoy digests, no Disclosure is sent to the Holder, i.e., the Holder will
+see digests that do not correspond to any Disclosure. See
+(#decoy_digests_privacy) for additional privacy considerations.
 
-Being JSON, an object in an SD-JWT payload MAY contain key-value pairs where the value is another object. In SD-JWT, the Issuer decides for each key individually, on each level of the JSON, whether the key should be selectively disclosable or not. This choice can be made on each level independent from whether keys higher in the hierarchy are selectively disclosable.
+To ensure readability and replicability, the examples in this specification do
+not contain decoy digests unless explicitly stated.
 
-For any selectively disclosable claim, the `_sd` key containing the digest value MUST be included in the SD-JWT at the same level as the original claim. It follows that the `_sd` key MAY appear multiple times in an SD-JWT. It MAY even appear within Disclosures.
+## Note on Nested Data in SD-JWTs {#nested_data}
+
+Being JSON, an object in an SD-JWT payload MAY contain key-value pairs where the value is another object or objects MAY be elements in arrays. In SD-JWT, the Issuer decides for each claim individually, on each level of the JSON, whether the claim should be selectively disclosable or not. This choice can be made on each level independent from whether keys higher in the hierarchy are selectively disclosable.
+
+From this it follows that the `_sd` key containing digests MAY appear multiple
+times in an SD-JWT, and likewise, there MAY be multiple arrays within the
+hierarchy with each having selectively disclosable elements. Digests of
+selectively disclosable claims MAY even appear within other Disclosures.
 
 The following examples illustrate some of the options an Issuer has. It is up to the Issuer to decide which option to use, depending on, for example, the expected use cases for the SD-JWT, requirements for privacy, size considerations, or ecosystem requirements.
```

Comment:
```suggestion
The following examples illustrate some of the options an Issuer has. It is up to the Issuer to decide which option to use, depending on, for example, the expected use cases for the SD-JWT, requirements for privacy, size considerations, or ecosystem requirements. For full examples with nested structures, see (#example-simple_structured) and (#example-complex-structured-sd-jwt).
```

I am a little confused why we need (#example-complex-structured-sd-jwt) if we have these 3 options. or I am a little confused why we need these three options if we have (#example-complex-structured-sd-jwt). but guess both are useful..

Reply:
The purpose of (#example-complex-structured-sd-jwt) alias `examples/complex_ekyc/user_claims.json` is to show *really* deeply nested objects. This section here shows a relatively simple structure to explain the basics.